jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/DefaultCredentialRetrievers.java

```diff
@@ -121,24 +123,24 @@ public DefaultCredentialRetrievers setCredentialHelper(@Nullable String credenti
     if (credentialHelper != null) {
       // If credential helper contains file separator, treat as path; otherwise treat as suffix
       if (credentialHelper.contains(FileSystems.getDefault().getSeparator())) {
-        if (Files.exists(Paths.get(credentialHelper))) {
-          credentialRetrievers.add(
-              credentialRetrieverFactory.dockerCredentialHelper(credentialHelper));
-        } else {
+        if (!Files.exists(Paths.get(credentialHelper))) {
           throw new FileNotFoundException(
               "Specified credential helper was not found: " + credentialHelper);
         }
+        credentialRetrievers.add(
+            credentialRetrieverFactory.dockerCredentialHelper(credentialHelper));
       } else {
+        String suffix = credentialHelper; // not path; treat as suffix
         credentialRetrievers.add(
-            credentialRetrieverFactory.dockerCredentialHelper(
-                DockerCredentialHelper.CREDENTIAL_HELPER_PREFIX + credentialHelper));
+            credentialRetrieverFactory.dockerCredentialHelper("docker-credential-" + suffix));
       }
     }
     if (inferredCredentialRetriever != null) {
       credentialRetrievers.add(inferredCredentialRetriever);
     }
     credentialRetrievers.add(credentialRetrieverFactory.dockerConfig());
-    credentialRetrievers.add(credentialRetrieverFactory.inferCredentialHelper());
+    credentialRetrievers.add(credentialRetrieverFactory.wellKnownCredentialHelper());
+    credentialRetrievers.add(credentialRetrieverFactory.googleApplicationDefaultCredentials());
```

Comment:
Shouldn't `googleApplicationDefaultCredentials()` come before `wellKnownCredentialHelper()`?

Reply:
I think ADC should be tried as a last resort. For example, the default GCE service account has a project editor role by default, but you can create instances with different roles. So most of the time, Jib will work out of the box, but ADC doesn't guarantee pushing to GCR will succeed.

Reply:
Makes sense.